### ISSUE TITLE
How to allow control plane request in strict NetworkPolicy mode?

### ISSUE THREAD

**Issue by weijiany** [aws/aws-network-policy-agent#281]
**What happened**:

I want to use strict NetworkPolicy to restrict my network in EKS cluster, at the same time I use some 3rd applications that have admission controller, they were called by control plane, and these requests will be failed due to I can't configure an ingress to allow control plane access it. Because the control plane is managed by AWS, I couldn't find pods in my EKS cluster, and I couldn't use a label selector to match them.

**Environment**:
- Kubernetes version: v1.30.0-eks-036c24b
- CNI Version: v1.18.2-eksbuild.1

Do you have some thoughts about this question? 🤔

**Comment by weijiany** [comment]
Hi this, any updates about this question?

**Comment by achevuru** [comment]
@weijiany  You want to allow requests coming in from APIServer? Why not just allow the endpoint IPs behind the `Kubernetes` service? I believe that should help with your use case....

**Comment by weijiany** [comment]
I think the endpoint IPs will be rotate when the api server restart. I don't know weather this is correct.

**Comment by achevuru** [comment]
Yes, that is accurate but there is unfortunately no easy way to select API Server (control plane) IPs via pod selectors in EKS right now. You would've to periodically check for any potential changes and update the NP. Another way is to allow the subnets you passed in for create cluster API (would make sense if you're using separate subnets for cluster create and nodes/instances). If you want to enable `egress` access to APIServer from pods then you can just allow `.1` IP of your cluster's Service VIP CIDR but that will not work for ingress traffic.

**Comment by weijiany** [comment]
I got. Now, we have already allowed 0.0.0.0 in admission controller's svc rather than vpc subnets CIDR, we maybe change them in the feature.

Thanks for your help. @achevuru
